gaminatorium-main/application/src/test/java/eu/gaminatorium/user/UserServiceTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    UserService userService;
+
+    private User user;
+
+    private UserDto userDto;
+
+    private User user2;
+
+    @BeforeEach
+    void setUp() {
+        user = User.builder()
+                .userName("User Name")
+                .email("user@gaminatoriu.pl")
+                .password("Password")
+                .build();
+
+        userDto = UserDto.builder()
+                .userName(user.getUserName())
+                .email(user.getEmail())
+                .password(user.getPassword())
+                .build();
+
+        user2 = User.builder()
+                .userName("User2")
+                .email("user2@gaminatoriu.pl")
+                .password("Password")
+                .build();
+    }
+
+    @Test
+    void shouldReturnUserDtoList() {
+        Page<User> userPage = new PageImpl<>(List.of(user, user2));
+
+        when(userRepository.findAllBy(any())).thenReturn(userPage);
+
+        List<UserDto> result = userService.findAll(any());
+
+        assertNotNull(result);
+        assertEquals(2, result.size());
+        assertEquals(user.getUserName(), result.getFirst().getUserName());
+    }
+
+    @Test
+    void shouldCountAllUsers() {
+        List<User> users = List.of(user, user2);
+        when(userRepository.countAllBy()).thenReturn(users.size());
+
+        long result = userService.countAllUsers();
+
+        assertEquals(users.size(), result);
+    }
+
+    @Test
+    void shouldReturnUserIfExist() {
+        when(userRepository.findById(anyLong())).thenReturn(Optional.ofNullable(user));
+
+        Optional<UserDto> result = userService.getUserById(anyLong());
+
+        assertTrue(result.isPresent());
+        assertEquals(result.get().getUserName(), user.getUserName());
+    }
+
+    @Test
+    void shouldReturnNullIfUserDoseExist() {
+        when(userRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        Optional<UserDto> result = userService.getUserById(anyLong());
+
+        assertFalse(result.isPresent());
+    }
+
+    @Test
+    void shouldCallUserRepository() {
+        userRepository.deleteById(anyLong());
+
+        verify(userRepository, times(1)).deleteById(anyLong());
+    }
+
+    @Test
+    void shouldUpdateUserIfExist() {
+        when(userRepository.findById(anyLong())).thenReturn(Optional.ofNullable(user));
+        when(userRepository.save(any(User.class))).thenReturn(user);
+
+        Optional<UserDto> result = userService.updateUser(anyLong(), userDto);
+
+        assertTrue(result.isPresent());
+    }
+
+    @Test
+    void shouldReturnEmptyIfUserDoseExist() {
+        when(userRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        Optional<UserDto> result = userService.updateUser(anyLong(), userDto);
+
+        assertFalse(result.isPresent());
```

Comment:
Wydaje mi się że lepiej być bardziej precyzyjnym i dać co ma być zamiast dawać czego ma nie być jeżeli oczekujemy konkretnego wyniku (Empty). 
```suggestion
        assertTrue(result.isEmpty());
```